src/figuresClasses.ts

```diff
@@ -1,19 +1,73 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string,
+  color: string,
+  getArea: () => number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c].sort((n1, n2) => n2 - n1);
+    const isPositiveSides = sides.some((el) => el <= 0);
+    const isInvalidSide = sides[0] >= sides[1] + sides[2];
+
+    if (isPositiveSides || isInvalidSide) {
+      throw new Error(`Fucking ${this.shape} Error`);
```

Comment:
🔥

Reply:
let's rewrite error text to users friendly one